app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result_list = [Person(name=person["name"], age=person["age"])
+                   for person in people]
+
+    for person in people:
+        if person.get("wife"):
+            main_person = Person.people[person["name"]]
+            wife_person = Person.people[person["wife"]]
+            main_person.wife = wife_person
+        elif person.get("husband"):
+            main_person = Person.people[person["name"]]
+            husband_person = Person.people[person["husband"]]
```

Comment:
Similarly, before accessing `Person.people[person['husband']]`, ensure that the husband exists in the `people` dictionary to avoid a potential KeyError.